### PR TITLE
feat: Adds option for restricting billing to admins

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

[Related request](https://posthog.slack.com/archives/C043VJ93L3B/p1713360019533869)

You might not want anyone in your org to be able to see and modify billing info

## Changes

* Adds an option to restrict to admins
* Adds billing to the settings area (I always go their first and get confused why it isn't there)
* When enabled restricts the billing area and specifically restricts any things that can modify billing from the api

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by raquelmsmith — COMMENTED** [review]
Mostly just wondering if we want to assume that admin-only is the correct and only permission we should offer here. If so, then this is fine. But if we think people might want billing to be owner-only, then this needs some tweaking. I'd prob go for the latter (tweaking) as it'll be more flexible for the (imo) inevitable requests we will get from enterprise customers about this.

File: ee/api/billing.py [hunk 1]
```diff
@@ -65,6 +75,10 @@ def patch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
             raise Exception("There is no license configured for this instance yet.")
 
         org = self._get_org_required()
+        if not self._can_manage_billing(request, org):
+            # Don't show the portal URL if the user doesn't have access to billing
+            raise PermissionDenied("Your organization restricts access to billing information to admins only.")
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply ee/api/billing.py:80]: Sounds good but sadly I've run out of steam for this PR 😅 feel free to approve, take it over or leave it and maybe I'll come back around next week

**Review by raquelmsmith — APPROVED** [review]
I'll approve because I don't have the bandwidth to take this on right now, and because this is good and better than nothing.

We will prob need to extend this in the future for owners only, or a special billing role.

@benjackwhite can you confirm that the upgrade flow (mostly the redirect) still worked properly with changing the billing URL?

**Comment by benjackwhite** [comment]
> @benjackwhite can you confirm that the upgrade flow (mostly the redirect) still worked properly with changing the billing URL?

I didn't change the URL, just added billing to settings as well. (More ways for people to see billing doesn't feel like a bad thing)

**Pull request closed without merging**